plugins/menubar/linux/src/menubar_plugin.cc

```diff
@@ -0,0 +1,160 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <menubar/menubar_plugin.h>
+
+#include <gtk/gtk.h>
+#include <stdio.h>
+
+#include "../../common/channel_constants.h"
+
+static constexpr char kWindowTitle[] = "Flutter Menubar";
+
+namespace plugins_menubar {
+using flutter_desktop_embedding::JsonMethodCall;
+using flutter_desktop_embedding::MethodResult;
+
+MenubarPlugin::MenubarPlugin()
+    : JsonPlugin(kChannelName, false) {}
+
+MenubarPlugin::~MenubarPlugin() {}
+
+// Class containing the implementation of the Menubar widget. This is currently
+// a floating GTK window, separate from the main app. This is not the optimal
+// solution.
+class MenubarPlugin::Menubar {
+ public:
+  explicit Menubar(MenubarPlugin *parent) {
+    menubar_window_ = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_position(GTK_WINDOW(menubar_window_), GTK_WIN_POS_CENTER);
+    gtk_window_set_default_size(GTK_WINDOW(menubar_window_), 300, 50);
+    gtk_window_set_title(GTK_WINDOW(menubar_window_), kWindowTitle);
+
+    GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+    gtk_container_add(GTK_CONTAINER(menubar_window_), vbox);
+
+    menubar_ = gtk_menu_bar_new();
+    gtk_box_pack_start(GTK_BOX(vbox), menubar_, FALSE, FALSE, 0);
+  }
+  virtual ~Menubar() {
+    if (menubar_window_) {
+      gtk_widget_destroy(menubar_window_);
+      gtk_widget_destroy(menubar_);
+    }
+  }
+
+  // Gets the top level menubar widget.
+  GtkWidget *GetRootMenuBar() { return menubar_; }
+
+  // Triggers an action once a menubar item has been selected.
+  static void MenuItemSelected(GtkWidget *menuItem, gpointer *data) {
+    auto plugin = reinterpret_cast<MenubarPlugin *>(data);
+
+    plugin->InvokeMethod(kMenuItemSelectedCallbackMethod,
+                         std::stoi(gtk_widget_get_name(menuItem)));
+  }
+
+  // Creates the menu items heirarchy from a given Json object.
+  void SetMenuItems(const Json::Value &root,
+                    flutter_desktop_embedding::Plugin *plugin,
+                    GtkWidget *parentWidget) {
+    if (root.isArray()) {
+      // This is the base of Json tree. It's not a menu item itself, so there's
+      // no need to create a widget.
+      unsigned int counter = 0;
+      while (counter < root.size()) {
+        if (root[counter].isObject()) {
+          SetMenuItems(root[counter], plugin, parentWidget);
+          counter++;
+        }
+      }
+      gtk_widget_show_all(menubar_window_);
+
+      return;
+    }
+
+    if ((root[kLabelKey]).isString()) {
```

Comment:
nit: Might want to remove the parens for consistency here (or add them in the other places that check for things like `isBool` etc).